src/py/flwr/client/app.py

```diff
@@ -692,3 +749,13 @@ def run_clientappio_api_grpc(
     log(INFO, "Starting Flower ClientAppIo gRPC server on %s", address)
     clientappio_grpc_server.start()
     return clientappio_grpc_server, clientappio_servicer
+
+
+def is_port_free(host: str, port: int) -> bool:
+    """Return a boolean indicating whether a specific port is free (not in use) on the
+    specified host."""
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+        # Attempt to connect to the specified host and port
+        result = sock.connect_ex((host, port))
+        # If connect_ex returns 0, the port is in use; otherwise, it's free
+        return result != 0
```

Comment:
Do we really need this? Or can we just try to start the gRPC server and rely on it failing if the port is used?

Reply:
removed since it is no longer used (see comment below)